File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int deepestLeavesSum(TreeNode root) {
        if(root==null)return 0;
        Queue<TreeNode> bfsQueue = new ArrayDeque<>();
        int deepSum=0;
        bfsQueue.add(root);
        while(!bfsQueue.isEmpty()){
            deepSum=0;
            int size = bfsQueue.size();
            for(int i=0; i<size; i++){
                TreeNode node = bfsQueue.poll();
                if(node.left==null && node.right==null)deepSum += node.val;
                if(node.left!=null)bfsQueue.add(node.left);
                if(node.right!=null)bfsQueue.add(node.right);
            }
        }
        return deepSum;
    }
}

File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def deepestLeavesSum(self, root: Optional[TreeNode]) -> int:
        level_q = deque()
        level_q.append(root)
        while len(level_q)>0:
            level_size = len(level_q)
            level_sum, last_level = 0, True
            for _ in range(level_size):
                node = level_q.popleft()
                level_sum += node.val
                if node.left:
                    last_level = False
                    level_q.append(node.left)
                if node.right:
                    last_level = False
                    level_q.append(node.right)
            if last_level:
                return level_sum
        return -1